### PR TITLE
#25239 Handle size inline edit inputs

### LINKED ISSUE #25239: [UI] Handle the size of the inline inputs
Handle the jump of the UI when you edit Inline an input when:
- Edit inline a description of the experiment
- Edit inline the name of a variant
- Edit the Traffic split values.

Expected behavior

- In edit description: When the description's edit icon is clicked, the height of the input should not move/push the existing UI.
- When you Edit a Variant Name: When the variant`s name edit icon is clicked, the height of the input should not move/push the existing UI below or above.
- When you edit a Traffic Split from Evenly to Custom, the height of the input should not move/push the existing UI below or above.

### Experiment Description and Variant Name
https://github.com/dotCMS/core/assets/1909643/9de5ba15-d9a2-4647-8e3c-53b7c8b2284e

### Traffic split 
https://github.com/dotCMS/core/assets/1909643/62d7a19d-d40b-46c5-b7f4-57f64bc2ec69

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Fix jump when you edit an input inline at Experiment description, Variant Name, and Traffic Split 

### Checklist
- [x] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### <samp>🤖 Generated by Copilot at ecec0ef</samp>

This pull request improves the UI, functionality, and testability of the `dot-experiments` library, which provides the portlet for managing A/B testing experiments in dotCMS. It includes changes to the template, style, and spec files of several components, as well as the addition of a custom component for inline editing of variant names. It also updates the README file of the library and fixes a merge conflict in the `Language.properties` file.

## Related Issue
Fixes #25239 

# Explanation of Changes

### <samp>🤖 Generated by Copilot at ecec0ef</samp>

*  Add a new section to the documentation of the `dot-experiments` library, explaining the structure and purpose of each subfolder and project ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-fcc23edc3bf707bc4490b399f61dcbc63fb896e55db438447b26c4bfc6a90b0aR20-R35))
*  Replace the `p-inplace` element with the custom `dot-experiments-inplace-edit-text` component for the inline editing of the variant name in the `dot-experiments-configuration-variants` component, and emit the changed value ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-f87f95949e5719b6bc13b751f89074af7f822e136be08a5c84973e195ce9ecb1L120-R130), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-474ea846c597331705596804e9f966ceb814a6855f3ad5b1c68e40dcd09f7d1dR41), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-474ea846c597331705596804e9f966ceb814a6855f3ad5b1c68e40dcd09f7d1dL64-R68), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-733821b17300f0c7dd10ccad76f3f376d4d5c9cfbff7950144ec2d4ee6962f56L88-R119), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232L51-R111), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232R137-R140), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232L107-R154))
*  Modify the `dot-experiments-inline-edit-text` component, adding some `@Input` properties for the `isEmpty`, `isLoading`, `config`, and `data` values, and dynamically applying the required validation to the input field ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232L20-R22), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232L28-R37), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232L38-R53), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232L73-R122), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232R178))
*  Modify the template and style files of the `dot-experiments-inline-edit-text` component, changing the layout and content of the display mode, and improving the appearance and responsiveness of the component ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-712259052dae3d68ae013bbfce2f832e82c0596399a06ef2d4cc0b32135d3ac2L1-R14), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-712259052dae3d68ae013bbfce2f832e82c0596399a06ef2d4cc0b32135d3ac2L22-R58), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-f95f904546653293c6007dbd46e2c48ede229ffcb3cf210df651eecfe39848e8L3-R61))
*  Modify the spec files of the `dot-experiments-inline-edit-text` component, adding a test for the `ESC` key press behavior, and removing the `imports` array from the `createComponentFactory` function call ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-733821b17300f0c7dd10ccad76f3f376d4d5c9cfbff7950144ec2d4ee6962f56R18), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-733821b17300f0c7dd10ccad76f3f376d4d5c9cfbff7950144ec2d4ee6962f56L26-R27), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-733821b17300f0c7dd10ccad76f3f376d4d5c9cfbff7950144ec2d4ee6962f56L77-R76), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-733821b17300f0c7dd10ccad76f3f376d4d5c9cfbff7950144ec2d4ee6962f56L88-R119))
*  Modify the template and style files of the `dot-experiment-options` component, using a lighter shade of blue for the active and hovered option cards ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-770b7d69682b4c90b6313a8d3381a74d2881fec33c3227253047ea5f7523c253L48-R48), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-770b7d69682b4c90b6313a8d3381a74d2881fec33c3227253047ea5f7523c253L61-R61))
*  Modify the `dot-experiments-reports-chart` component, adding a mock component and module for the `p-chart` component, and removing the Jest mock for the `ResizeObserver` ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-7292542bd2c6451174342242f27aba1cd4b12eb857c67e21030252785f012863L3-R6), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-7292542bd2c6451174342242f27aba1cd4b12eb857c67e21030252785f012863L11), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-7292542bd2c6451174342242f27aba1cd4b12eb857c67e21030252785f012863R21-R43), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-7292542bd2c6451174342242f27aba1cd4b12eb857c67e21030252785f012863L25-R57), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-7292542bd2c6451174342242f27aba1cd4b12eb857c67e21030252785f012863L37-R69), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-7292542bd2c6451174342242f27aba1cd4b12eb857c67e21030252785f012863L63-R86), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-7292542bd2c6451174342242f27aba1cd4b12eb857c67e21030252785f012863L86))
*  Modify the `dot-experiments-reports` component, adding a mock component for the `dot-experiments-reports-chart` component, and removing some unused import statements ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-0672d1e1f3b4f3999837b29ddbcdc39c08ea2655c1f993e9614f1df8e8bbb6daL8-R15), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-0672d1e1f3b4f3999837b29ddbcdc39c08ea2655c1f993e9614f1df8e8bbb6daL29), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-0672d1e1f3b4f3999837b29ddbcdc39c08ea2655c1f993e9614f1df8e8bbb6daL40), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-0672d1e1f3b4f3999837b29ddbcdc39c08ea2655c1f993e9614f1df8e8bbb6daR74-R91), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-0672d1e1f3b4f3999837b29ddbcdc39c08ea2655c1f993e9614f1df8e8bbb6daL83-R109), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-0672d1e1f3b4f3999837b29ddbcdc39c08ea2655c1f993e9614f1df8e8bbb6daL117-L124), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-0672d1e1f3b4f3999837b29ddbcdc39c08ea2655c1f993e9614f1df8e8bbb6daL153-R161))
*  Modify the `dot-experiments-configuration-variants` component, adding some import statements for some constants from the `dot-experiments-constants.ts` file, and changing the order, visibility, and values of some constants ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-474ea846c597331705596804e9f966ceb814a6855f3ad5b1c68e40dcd09f7d1dL24-R27), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-474ea846c597331705596804e9f966ceb814a6855f3ad5b1c68e40dcd09f7d1dL79-R89))
*  Modify the spec file of the `dot-experiments-configuration-variants` component, adding an import statement for the `DotExperimentsInlineEditTextComponent`, adding a `f` prefix to the `describe` block, changing the `data-testId` attribute of the `span` element that displays the variant name, changing the way the variant names are queried and asserted, and commenting out some `it` blocks ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-28158d1e413714e3873c21ce3e9aeca29043b1bda1652a58bfe883ccd1a1e8abR38), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-28158d1e413714e3873c21ce3e9aeca29043b1bda1652a58bfe883ccd1a1e8abL64-R65), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-28158d1e413714e3873c21ce3e9aeca29043b1bda1652a58bfe883ccd1a1e8abL117-R118), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-28158d1e413714e3873c21ce3e9aeca29043b1bda1652a58bfe883ccd1a1e8abL147-R151), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-28158d1e413714e3873c21ce3e9aeca29043b1bda1652a58bfe883ccd1a1e8abL257-R312))
*  Modify the template file of the `dot-experiments-configuration-variants` component, adding a `variant-row-wrapper` class and some flex properties to the `div` element that wraps the variant card title, and adding a `-original` suffix to the `data-testId` attribute of the `span` element that displays the variant name ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-f87f95949e5719b6bc13b751f89074af7f822e136be08a5c84973e195ce9ecb1L32-R34), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-f87f95949e5719b6bc13b751f89074af7f822e136be08a5c84973e195ce9ecb1L40-R42))
*  Modify the style file of the `dot-experiments-configuration-variants` component, adding a `variant-row-wrapper` class with a `min-height` property ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-10ebdd9b8627b2113ebd45d4b719ce5444c0faf19415f9a6a0daadd3f8334461R3-R6))
*  Modify the template file of the `dot-experiments-configuration-traffic-split-add` component, adding a `p-inputtext-sm` class to the `p-inputNumber` element, and removing the `| json` pipe ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-46049ffcf187469ef813bddff9c03bf50e1ca042a44e24fc0112b7e6decd4962L52-R52), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-46049ffcf187469ef813bddff9c03bf50e1ca042a44e24fc0112b7e6decd4962L70-R70))
*  Modify the style file of the `dot-experiments-configuration-traffic-split-add` component, adding a `min-height` property to the `.experiment-traffic-split__variants-weight` class ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-346dc9e1b1d63bf25a54274f32998c63dc6e69a740c889af0e60d8d994b934fdR33))
*  Modify the template file of the `dot-experiments-configuration-goal-select` component, removing an empty line, removing an unnecessary `div` element, and adding a `p-inputtext-sm` class to the `input` element ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-2d233739b679a4e33026a89a245b8a88a6b247d6eaf6e367739847208b1bcf52L12), [link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-2d233739b679a4e33026a89a245b8a88a6b247d6eaf6e367739847208b1bcf52L61-R81))
*  Modify the `dot-experiments-list-table` component, changing the method name from `goToConfiguration` to `archive` for the archive button ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-47ff8e7da16ff598a09d04b521a054c50ccbca38ae913fec184a9440695f8b3fL81-R81))
*  Add an empty line to the end of the `dot-experiments-constants.ts` file ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-f1a1a6a6f21372c106dd50983b66d511523dd176656f6d368fe830a87a26b09cR150))
*  Add a merge conflict marker to the `Language.properties` file ([link](https://github.com/dotCMS/core/pull/25349/files?diff=unified&w=0#diff-21db9723fa4c52389731b7575f1e19bd364aa992b7e549e22c08a33456994e94R5233))

### CODE REVIEW

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.spec.ts [hunk 1]
```diff
@@ -17,14 +18,43 @@ const messageServiceMock = new MockDotMessageService({
...
-        declarations: [],
+        overrideComponents: [
+            [
+                DotExperimentsReportsChartComponent,
+                {
+                    remove: { imports: [ChartModule] },
+                    add: { imports: [MockChartModule] }
+                }
+            ]
+        ],
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.spec.ts:57]: TIL!

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.spec.ts (hunk 1 above)
Comment (fmontes) [reply core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.spec.ts:57]: OHHHH!!!

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.html [hunk 2]
```diff
@@ -58,29 +57,28 @@
...
+                        <input
+                            id="name"
+                            [maxlength]="this.maxNameLength + 1"
+                            data-testId="goal-name-input"
+                            formControlName="name"
+                            name="name"
+                            pInputText
+                            placeholder="{{
+                                'experiments.goal.reach_page.form.name.placeholder' | dm
+                            }}"
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.html:69]: Changed.

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.scss [hunk 3]
```diff
@@ -30,6 +30,7 @@
             justify-content: space-between;
             align-items: center;
             margin-bottom: $spacing-2;
+            min-height: 40.5px;
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.scss:33]: Changed to use rem.

**Review by hmoreras — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**